lib/net/ldap/connection.rb

```diff
@@ -36,7 +36,7 @@ def prepare_socket(server)
     encryption = server[:encryption]
 
     @conn = socket
-    setup_encryption encryption if encryption
+    setup_encryption({ verify_host: server[:connected_host] }.merge(encryption)) if encryption
```

Comment:
@tarcieri, sorry I'm not really much of a ruby dev, so I may misunderstand a bit. I assume any non nil value in a conditional returns false... so users must explicitly enable verification in config with `:encryption = { :method => :simple_tls, :tls_options => { :verify_host => true } }`?

If I understand correctly, `:encryption => { :method => :simple_tls, :tls_options => { :verify_mode => OpenSSL::SSL::VERIFY_PEER, :tls_options[:verify_host] => false} }` should imply checking that the cert is at least signed by a trusted CA, but allowing for a mismatch?

What happens if the user has simply put `{ :method => :simple_tls, :tls_options => { :verify_mode => OpenSSL::SSL::VERIFY_PEER`}`?

```
+    if tls_options[:verify_host]
 +      # This raises OpenSSL::SSL::SSLError if hostname verification fails
 +      conn.post_connection_check(tls_options[:verify_host])
```
The match function will not run if tls_options[:verify_host] is nil? Hence I thought that your pull doesn't do hostname validation unless the user explicitly configures it.